orc8r/cloud/go/swagger/combine.go

```diff
@@ -0,0 +1,197 @@
+/*
+ Copyright 2020 The Magma Authors.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package swagger
+
+import (
+	"regexp"
+	"sort"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+// Combine multiple Swagger specs, giving precedence to the "common" spec.
+// First error contains merge warnings for overwritten fields and
+// incompatible Swagger specs which will be explicitly ignored in the merging
+// process.
+// This custom-built functionality mirrors the "official" implementation.
+// See: https://github.com/go-openapi/analysis/blob/master/mixin.go
+func Combine(common string, spec []string) (string, error, error) {
+	warnings := &multierror.Error{}
+	specs, commonSpec, errs := unmarshalToSwagger(common, spec)
+	if errs != nil {
+		multierror.Append(warnings, errs)
+	}
+
+	combinedSpec, errs := combine(commonSpec, specs)
```

Comment:
Nit following from the previous comment: here + in tests + in tool code/tests: let's just use `common` rather than `commonSpec`, since adding `Spec` is superfluous given that the object type is `SwaggerSpec` (meaning `Spec` is inferred from the type)